src_py/sprite.py

```diff
@@ -66,41 +66,37 @@
 
 """
 
-##todo
-## a group that holds only the 'n' most recent elements.
-## sort of like the GroupSingle class, but holding more
-## than one sprite
-##
-## drawing groups that can 'automatically' store the area
-## underneath so they can "clear" without needing a background
-## function. obviously a little slower than normal, but nice
-## to use in many situations. (also remember it must "clear"
-## in the reverse order that it draws :])
-##
-## the drawing groups should also be able to take a background
-## function, instead of just a background surface. the function
-## would take a surface and a rectangle on that surface to erase.
-##
-## perhaps more types of collision functions? the current two
-## should handle just about every need, but perhaps more optimized
-## specific ones that aren't quite so general but fit into common
-## specialized cases.
+# TODO:  a group that holds only the 'n' most recent elements.
+#        sort of like the GroupSingle class, but holding more
+#        than one sprite
+#
+# drawing groups that can 'automatically' store the area
+# underneath so they can "clear" without needing a background
+# function. obviously a little slower than normal, but nice
+# to use in many situations. (also remember it must "clear"
+# in the reverse order that it draws :])
+#
+# the drawing groups should also be able to take a background
+# function, instead of just a background surface. the function
+# would take a surface and a rectangle on that surface to erase.
+#
+# perhaps more types of collision functions? the current two
+# should handle just about every need, but perhaps more optimized
+# specific ones that aren't quite so general but fit into common
+# specialized cases.
 
-import pygame
-from pygame import Rect
```

Comment:
I was wondering if it's worth binding Rect here for speed, but I guess anywhere it's used in a loop it should just bind it to a local first.

Reply:
:thumbs:

I guess performance tests for the sprite module would be useful for refactorings.

Reply:
If speed is relevant, then I think we should use Cython for the sprite classes instead of pure python - especially if we make `pygame.Rect` C-callable for Cython (I do not remember if it is already)

Reply:
I checked and it looks like Rect is bound to a local variable/function argument prior to any time it is used in a loop.